### ISSUE TITLE
registration actually fails if system clock is slightly more ahead from ASVZ server clock

### ISSUE THREAD

**Issue by lthiet** [jstiefel/asvz_bot#4]
hi, sorry for another post haha

i noticed that there is no confirmation or unexpected behaviour handling after the enroll was clicked. in this instance, i was greeted with an alert that says : "Der Anmeldebeginn liegt in der Zukunft - eine Anmeldung ist leider noch nicht möglich!".

maybe a fix would be to check for the presence of this element and retry until it disappears? or check that the green 'success' box is present before closing the script?

**Comment by jstiefel** [comment]
Yes, there are many cases which are currently not handled in this script. I was about to release a version 2 with several improvements and handling of special cases. However, since the enrollment process is changing, I'm waiting with it. This version is more of a template.

**Comment by lthiet** [comment]
I see. Looking forward to the release of V2 then! In the meantime, would you be so kind as to explain what changes you made in the high level to deal with this kind of situation so that I can implement it as a temporary makeshift patch in my forked version of your bot?

**Comment by jstiefel** [comment]
Sure, I added a lot of exception handling, like expanding the activity page, checking if the forwarding works, checking if the enrollment was successful (exactly what you wrote above with checking if the button is available), exchanging the waiting function, ...
